docs/user_guides/single_table/custom_constraints.rst

```diff
@@ -0,0 +1,239 @@
+
+-  ``table_data`` which contains all the information.
+-  ``column`` which is a an argument to represent the columns of interest.
+
+Now we can construct our functions freely, we write our methods
+with said arguments and be able to access ``'contractor'``.
+
+We first write our ``transform`` function as we have done previously:
+
+.. ipython:: python
+    :okwarning:
+
+    def transform(table_data, column):
+        base = 500
+        table_data[column] = table_data[column] / base
+        return table_data
+
+When it comes to defining ``reverse_transform``, we need to distinguish between
+contractors and non contractors, the operations are as follows:
+
+1. round values to four decimal points for contractors such that the end result will 
+   be two decimal points after multiplying the result with 500.
+2. round values to zero for employees that are not contractors such that the end
+   result will be a multiple of 500.
+
+.. ipython:: python
+    :okwarning:
+
+    def reverse_transform(table_data, column):
+        base = 500.
+        is_not_contractor = table_data.contractor == 0.
+        table_data[column] = table_data[column].round(4)
+        table_data[column].loc[is_not_contractor] = table_data[column].loc[is_not_contractor].round()
+        table_data[column] *= base
+        return table_data
+
+We now stich everything together and pass it to the model.
+
+.. ipython:: python
+    :okwarning:
+
+    constraint = CustomConstraint(
+        columns=['salary', 'annual_bonus'],
+        transform=transform, 
+        reverse_transform=reverse_transform
+    )
+
+    gc = GaussianCopula(constraints=[constraint])
+
+    gc.fit(employees)
+
+    sampled = gc.sample(10)
+
+
+When we view the ``sampled`` data, we should find that all the rows in the sampled 
+data have a salary that is a multiple of the base value with the exception
+of "contractor" records.
+
+.. ipython:: python
+    :okwarning:
+
+    sampled
+
+This style gives flexibility to access any column in the table while still operating on 
+a column basis.
+
+
+Can I write a ``CustomConstraint`` based on reject sampling?
+------------------------------------------------------------
+
+In the previous section, we defined our ``CustomConstraint`` using ``transform`` and 
+``reverse_transform`` functions. Sometimes, our constraints are not possible to implement 
+using these methods, that is when we rely on ``reject_sampling`` strategy. In ``reject_sampling`` 
+we need to implement an ``is_valid`` function that identifies which rows do not follow the 
+said constraint, in our case, which rows are not a multiple of the *base* unit.
+
+We can define ``is_valid`` according to the three styles mentioned in the previous section:
+
+1. function with ``table_data`` argument.
+2. function with ``column_data`` argument.
+3. function with ``table_data`` and ``column`` argument.
+
+Here is an example of how you would define ``is_valid`` for each one of the mentioned styles:
```

Comment:
I would say somewhere explicitly what the expected output is. (A pd.Series where every valid row is 1 and the rest are 0).

Reply:
I made a minor adjustment and made it *True* and *False*, I think that's easier to understand.